src/components/HdToggle.vue

```diff
@@ -0,0 +1,251 @@
+      type="button"
+      @click="toggleOpen"
+    >
+      {{ title }}
+
+      <HdIcon
+        :src="chevronIcon"
+        class="hd-toggle__control__icon"
+      />
+    </button>
+    <div
+      ref="body"
+      :style="{
+        maxHeight: `${maxHeight}px`,
+        transitionDuration: `${transitionDuration}ms`,
+      }"
+      class="hd-toggle__body"
+    >
+      <slot />
+    </div>
+  </div>
+</template>
+
+<script>
+import { OnResizeService } from 'homeday-blocks';
+import HdIcon from 'homeday-blocks/src/components/HdIcon.vue';
+import { chevronIcon } from 'homeday-blocks/src/assets/small-icons';
+
+export default {
+  name: 'HdToggle',
+  components: {
+    HdIcon,
+  },
+  props: {
+    title: {
+      type: String,
+      default: '',
+    },
+    canBeToggled: {
+      type: Boolean,
+      default: true,
+    },
+    open: {
+      type: Boolean,
+      default: true,
+    },
+    transitionDuration: {
+      type: Number,
+      default: 300,
+    },
+  },
+  data() {
+    return {
+      // bodyHeight is a string only initially, will be set to a Number
+      // as soon as it is mounted
+      bodyHeight: '100%',
+      isUsingMouse: false,
+      internalFocusRemoved: false,
+      chevronIcon,
+    };
+  },
+  computed: {
+    maxHeight() {
+      if (this.open) {
+        return this.bodyHeight;
+      }
+
+      return 0;
+    },
+  },
+  watch: {
+    open: {
+      handler(open) {
+        this.ensureBodyScrolledToTop();
+
+        this.$nextTick(() => {
+          if (open) {
+            this.enableInternalFocus();
+          } else {
+            this.disableInternalFocus();
+          }
+        });
+      },
+      immediate: true,
+    },
+  },
+  mounted() {
+    this.$nextTick(this.calculateBodyHeight);
+    this.addResizeEvents();
+  },
+  beforeDestroy() {
+    this.removeResizeEvents();
+  },
+  methods: {
+    // Encapsulated in a method to be able to mock it in the tests
+    getScrollHeight(el) {
+      return el.scrollHeight;
+    },
+    toggleOpen() {
+      if (this.canBeToggled === false) {
+        return;
+      }
+
+      this.$emit('toggle', !this.open);
+    },
+    calculateBodyHeight() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      this.bodyHeight = this.getScrollHeight(this.$refs.body);
+    },
+    ensureBodyScrolledToTop() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      this.$refs.body.scrollTop = 0;
+    },
+    disableInternalFocus() {
+      if (!this.$refs.body) {
+        return;
+      }
+
+      const $innerFocusableElements = this.$refs.body.querySelectorAll(
+        'a, button, [tabindex]',
+      );
+
+      if (!$innerFocusableElements.length) {
+        return;
+      }
+
+      Array.from($innerFocusableElements).forEach((el) => {
+        const currentTabIndex = el.getAttribute('tabindex');
+
+        if (currentTabIndex !== null) {
+          el.setAttribute('data-hd-tabindex-backup', currentTabIndex);
```

Comment:
`'data-hd-tabindex-backup'` can go in a constant as it's used in two other places.